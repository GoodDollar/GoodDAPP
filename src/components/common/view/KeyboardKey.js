// @flow
import React from 'react'
import { Platform, TouchableHighlight, View } from 'react-native'
import { withStyles } from '../../../lib/styles'
import Icon from './Icon'
import Text from './Text'

type KeyboardKeyProps = {
  keyValue: string,
  onPress: string => void,
}

const KeyboardKey = ({ keyValue, onPress, styles, theme }: KeyboardKeyProps) => {
  return (
    <TouchableHighlight
      activeOpacity={0.8}
      onPress={() => onPress(keyValue)}
      style={styles.key}
      underlayColor={theme.colors.lightGray}
    >
      {keyValue === 'backspace' ? (
        <View style={styles.backspaceButton}>
          <Icon name="backspace" color={theme.colors.darkGray} size={18} style={{ textAlign: 'center' }} />
        </View>
      ) : (
        <Text fontSize={20} fontWeight="bold" fontFamily={theme.fonts.slab}>
          {keyValue}
        </Text>
      )}
    </TouchableHighlight>
  )
}

const getStylesFromProps = ({ theme }) => {
  return {
    key: {
      display: 'flex',
      alignItems: 'center',
      flex: 1,
      padding: theme.sizes.default,
      ...Platform.select({
        web: { cursor: 'pointer' },
      }),
    },
    backspaceButton: {
      ...Platform.select({
        web: {
          backgroundRepeat: 'no-repeat',
          backgroundSize: 'contain',
          cursor: 'pointer',
        },
      }),
      height: 30,
      justifyContent: 'center',
      width: 40,
    },
  }
}

export default withStyles(getStylesFromProps)(KeyboardKey)
