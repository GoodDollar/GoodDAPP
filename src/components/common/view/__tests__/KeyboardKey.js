import React from 'react'
import renderer from 'react-test-renderer'
import KeyboardKey from '../KeyboardKey'

import { withThemeProvider } from '../../../../__tests__/__util__'

// Note: test renderer must be required after react-native.

describe('KeyboardKey', () => {
  const WrappedKeyboardKey = withThemeProvider(KeyboardKey)

  it('matches snapshot', async () => {
    let component

    await renderer.act(
      // eslint-disable-next-line require-await
      async () => (component = renderer.create(<WrappedKeyboardKey keyValue="1" onPress={() => {}} />)),
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
