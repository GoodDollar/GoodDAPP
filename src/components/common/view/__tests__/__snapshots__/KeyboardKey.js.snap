// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardKey matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-display-6koalj r-flex-13awgt0 r-padding-edyy15 r-touchAction-1otgn73 r-userSelect-lrvibr"
    onClick={[Function]}
    onContextMenu={[Function]}
    onKeyDown={[Function]}
    tabIndex="0"
  >
    <div
      className="css-text-901oao r-color-cddv6f r-fontFamily-6jnb14 r-fontSize-hiw28u r-fontWeight-b88u0q r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
      dir="auto"
    >
      1
    </div>
  </div>
</div>
`;
